packages/realm-react/src/RealmContext.ts

```diff
@@ -0,0 +1,109 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2024 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { createUseObject } from "./useObject";
+import { createUseQuery } from "./useQuery";
+import { createUseRealm } from "./useRealm";
+import { DynamicRealmProvider } from "./RealmProvider";
+
+export type RealmContext<RealmProvider = DynamicRealmProvider> = {
+  /**
+   * The Provider component that is required to wrap any component using
+   * the Realm hooks.
+   * @example
+   * ```
+   * const AppRoot = () => {
+   *   const syncConfig = {
+   *     flexible: true,
+   *     user: currentUser
+   *   };
+   *
+   *   return (
+   *     <RealmProvider schema={[Task, User]} path={"data.realm"} sync={syncConfig}>
+   *       <App/>
+   *     </RealmProvider>
+   *   )
+   * }
+   * ```
+   * @param props - The {@link Realm.Configuration} or {@link Realm} of the provider
+   * are set based on the options passed to `createRealmProvider`. When using a
+   * {@link Realm.Configuration}, individual config keys can be overridden when
+   * creating a `<RealmProvider>` by passing them as props. For example, to override
+   * the `path` config value, use a prop named `path` e.g., `path="newPath.realm"` an
+   * attribute of the same key.
+   */
+  RealmProvider: RealmProvider;
+  /**
+   * Returns the instance of the {@link Realm} opened by the `RealmProvider`.
+   * @example
+   * ```
+   * const realm = useRealm();
+   * ```
+   * @returns a realm instance
+   */
+  useRealm: ReturnType<typeof createUseRealm>;
+
+  /**
+   * Returns a {@link Realm.Collection} of {@link Realm.Object}s from a given type.
+   * The hook will update on any changes to any object in the collection
+   * and return an empty array if the collection is empty.
```

Comment:
> an empty array if the collection is empty

Is that true? I was in the understanding that it just returns the empty collection?

Reply:
this is a copy-paste from the current documentation, may need to check

Reply:
Hm yeah it does seem like this isn't really the case, it always return a collection and not an array. Not sure what the useful of that line is anyways so I think I'll just remove it